src/economy_func.h

```diff
@@ -49,4 +51,47 @@ static inline bool EconomyIsInRecession()
 	return _economy.fluct <= 0;
 }
 
+/**
+ * Scale a number by the inverse of the cargo scale setting, e.g. a scale of 25% multiplies the number by 4.
+ * @param num The number to scale.
+ * @return The number scaled by the inverse of the cargo scale setting, minimum of 1.
+ */
+static uint ScaleByInverseCargoScale(uint num)
+{
+	/* We might not need to do anything. */
+	if (_settings_game.economy.cargo_scale == 100) return num;
+
+	uint scaled;
+	if (_settings_game.economy.cargo_scale > 100) {
+		scaled = (num * 100) / _settings_game.economy.cargo_scale;
+	}
+	else {
+		scaled = (100 / _settings_game.economy.cargo_scale) * num;
```

Comment:
why are there two separate cases? it's the exact same formula but less precise, cargo_scale 51-100 will be exactly the same afaict.